### PR TITLE
Add groups filter to the users list

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Add the possibility to filter users by their group assignments.

Needs https://github.com/owncloud/ocis/pull/5533 for the server.

## Screenshot

image

## Related Issue

- Fixes https://github.com/owncloud/web/issues/8377

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by JammingBen** [comment]
Blocked until the server side implementation is finished: https://github.com/owncloud/ocis/issues/5487.

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue [hunk 1]
```diff
@@ -4,10 +4,17 @@
       id="users-filter"
       v-model="filterTerm"
       class="oc-ml-m oc-my-s"
-      :label="$gettext('Filter users')"
+      :label="$gettext('Search')"
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/components/Users/UsersList.vue:7]: Could we make it consistent, for groups and spaces as well?

File: packages/web-app-admin-settings/src/views/Users.vue [hunk 2]
```diff
@@ -2,7 +2,6 @@
   <div>
     <app-template
       ref="template"
-      :loading="loadResourcesTask.isRunning || !loadResourcesTask.last"
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/views/Users.vue:5]: why was this removed?

**Review by JanAckermann — CHANGES_REQUESTED** [review]

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/components/Users/UsersList.vue:7]: Good point!

File: packages/web-app-admin-settings/src/views/Users.vue (hunk 2 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/views/Users.vue:5]: It just moved from there to the current view, because the filter triggers the user loading. It should only re-load the users-list, not the entire page.

Comment (JanAckermann) [reply packages/web-app-admin-settings/src/views/Users.vue:5]: can we do the same with the other views and purge it from the app template?

**Review by JanAckermann — CHANGES_REQUESTED** [review]

File: packages/web-app-admin-settings/src/views/Users.vue (hunk 2 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/views/Users.vue:5]: Yes we should do that, but I'd rather do it in a separate PR to keep this PR here about the filter only.

File: packages/web-app-admin-settings/src/views/Users.vue [hunk 3]
```diff
@@ -156,6 +191,11 @@ export default defineComponent({
       return data
     })
 
+    const filterGroups = (groups) => {
+      const groupIds = groups.map((g) => g.id)
+      loadResourcesTask.perform(false, groupIds)
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/views/Users.vue:196]: loadUserTask instead pls

**Review by JanAckermann — CHANGES_REQUESTED** [review]

File: packages/web-app-admin-settings/src/views/Users.vue [hunk 4]
```diff
@@ -138,11 +163,21 @@ export default defineComponent({
...
 
-    const loadResourcesTask = useTask(function* (signal) {
-      const usersResponse = yield unref(graphClient).users.listUsers('displayName')
+    const loadResourcesTask = useTask(function* (signal, loadGroups = true, groupIds = null) {
+      const groupFilter = groupIds
+        ?.reduce((acc, id) => {
+          acc += `memberOf/any(m:m/id eq '${id}') and `
+          return acc
+        }, '')
+        .slice(0, -5)
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/views/Users.vue:172]: Would love to see join here

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Review by JanAckermann — APPROVED** [review]
LGTM :1st_place_medal:
